usr/src/cmd/zoneadmd/log.c

```diff
@@ -624,37 +669,36 @@ escape_json(const char *sbuf, int slen, char *dbuf, int dlen, int *scntp,
 			} else {
 				len = snprintf(append_buf, sizeof (append_buf),
 				    "\\u%04x", (int)(0xff & c));
-				assert(len < sizeof (append_buf));
+				VERIFY3S(len, <, sizeof (append_buf));
 			}
 			append = append_buf;
 			break;
 		}
 
-		len = strlcpy(&dbuf[i], append, dlen - i);
-		if (len >= dlen - i) {
+		len = strlen(append);
+
+		if (custr_append(dest, append) != 0) {
+			VERIFY3S(errno, ==, EOVERFLOW);
 			if (flushp != NULL) {
 				*flushp = B_TRUE;
 			}
 			break;
-		} else {
-			sbuf++;
-			i += len;
 		}
+
+		i += len;
+		sbuf++;
+
 		if (flushp != NULL && *flushp) {
 			break;
 		}
 	}
 
-	dbuf[i] = '\0';
-
-	*dcntp = i;
 	*scntp = sbuf - save_sbuf;
 
-	assert(*dcntp < dlen);
-	assert(*scntp <= slen);
+	VERIFY3U(*scntp, <=, slen);
 
-	/* Buffer is too full to append "\\n". Force a flush. */
-	if (flushp != NULL && i >= dlen - 2) {
+	/* Buffer is too full to append "\\n" + NUL. Force a flush. */
```

Comment:
I'm confused by this existing code and the new version. What appended newline?

Reply:
I'm not sure either -- I think this was all to try to force a flush if the escaped version was too large to fit into the output buffer. Since custr_t already handles that, I think the setting of flushp and breaking out of the loop of the custr_append() call fails should handle that. I'll update accordingly.